agent/discovery/sapsystem.go

```diff
@@ -26,28 +28,35 @@ func (d SAPSystemsDiscovery) GetId() string {
 	return d.id
 }
 
-func (d SAPSystemsDiscovery) Discover() error {
+func (d SAPSystemsDiscovery) Discover() (string, error) {
+	var sysNames string
 	systems, err := sapsystem.NewSAPSystemsList()
 
 	if err != nil {
-		return err
+		return "", err
 	}
 
 	d.SAPSystems = systems
 	for _, s := range d.SAPSystems {
+		sysNames = systems.GetSIDsString()
```

Comment:
By the way, you should move the `systems.GetSIDsString` out of the loop. It will always do the same